controller_manager/src/controller_manager.cpp

```diff
@@ -95,6 +88,39 @@ ControllerManager::ControllerManager(
   init_services();
 }
 
+void ControllerManager::init_resource_manager()
+{
+  std::string robot_description = "";
+  get_parameter("robot_description", robot_description);
+  if (robot_description.empty())
+  {
+    throw std::runtime_error("Unable to initialize resource manager, no robot description found.");
+  }
+
+  // TODO(destogl): manage this when there is an error - CM should not die because URDF is wrong...
+  resource_manager_->load_urdf(robot_description);
+
+  using lifecycle_msgs::msg::State;
+
+  std::vector<std::string> autoconfigure_components = {""};
+  get_parameter("autoconfigure_components", autoconfigure_components);
+  rclcpp_lifecycle::State inactive_state(
```

Comment:
Because we need to create a state each time with ID and name. Should we create a bunch of defines for the states?

Reply:
Because the upstream library does not defines standard states fully, at least not with the ID or names. We could create defines for this or should we directly try to add this into rclcpp?

Reply:
Also it is not possible to set this const because of the underlying logic where no copy of state is done but in place changes are added if ID is not provided... Will comment more in corresponding PR.